### PR TITLE
feat/bindable

### PR DESCRIPTION
**Pull request by NNikolov1415**
3.1.0 Change log:
- Implemented differentiation between `.sink` and `.sinkAndFire` to enable functionality equivalent to `.bind` and `.bindAndFire` available through TwoWayBondage.
- Implemented `Bindable` protocol and conformance to it for `UIControl` subclasses and for `UITextView`.
- Implemented `.oneWayBind(with bindable:)` and `.twoWayBind(with bindable:)` for `Observable` to enable functionality similar to `bind(with observable:)` available through TwoWayBondage. 
- Changes to the Bindable UIControl's value (e.g. UITextField's `text`) update the bound observable's value. With `.twoWayBind(with bindable:)`, changes to the bound observable's value update the Bindable UIControl's value in turn. 
- When using `twoWayBind`, `animateUpdates` boolean value can be passed (default = true), which allows the control to animate the update if it's supported.
- Added Notification Names to `Notification.Name+Custom` to enable `.oneWayBind(with bindable:)` and `.twoWayBind(with bindable:)`.
- Updated README.md with an example of  `.oneWayBind(with bindable:)` usage.

### CODE REVIEW

File: SFBaseKit/Utils/Observable.swift [hunk 1]
```diff
@@ -32,12 +32,42 @@ public final class Observable<T> {
...
+            .dropFirst()
             .receive(on: dispatchQueue)
             .sink(receiveValue: receiveValue)
             .store(in: &cancellables)
     }
+    
+    /// Binds the observable value to the values received from a Bindable's Publisher.
+    /// - Parameters:
+    ///   - bindable: A bindable object to receive values from.
+    public func sink<B: Bindable>(with bindable: B, on dispatchQueue: DispatchQueue = .main) {
```
Comment (nProdanov) [inline SFBaseKit/Utils/Observable.swift]: I am wondering whether this function should be as overload or with a brand new name. I find for confusing to have two functions with same name, that do the opposite. :?

**Review by kostadinzamanov — APPROVED** [review]

**Review by m-sapunova — APPROVED** [review]

**Review by ODimitrovSF — APPROVED** [review]

**Review by sveyp — APPROVED** [review]

File: SFBaseKit/Utils/Observable.swift [hunk 2]
```diff
@@ -32,12 +32,49 @@ public final class Observable<T> {
...
         subject
+            .dropFirst()
             .receive(on: dispatchQueue)
             .sink(receiveValue: receiveValue)
             .store(in: &cancellables)
     }
+    
+    /// Binds the observable value to the values received from a Bindable's Publisher.
+    /// - Parameters:
+    ///   - bindable: A bindable object to receive values from.
```
Comment (stoikokolev) [inline SFBaseKit/Utils/Observable.swift:62]: I suggest we stick to one pattern when listing parameters.
_/// Documentation option one
/// - Parameters:
///   - firstParameterName:
///   - secondParameterName:_

_/// Documentation option two
/// - Parameter firstParameterName:
/// - Parameter secondParameterName:_ 

The two different options are used in the _Observable_ and _Bindable_ files. I also suggest to document all the parameters.

Comment (NNikolov1415) [reply SFBaseKit/Utils/Observable.swift:62]: Updated parameter documentation

**Review by viktor-georgiev — APPROVED** [review]

**Review by BoyanJakimov — APPROVED** [review]

**Review by stoikokolev — APPROVED** [review]

File: SFBaseKit/Utils/Observable.swift [hunk 3]
```diff
@@ -32,12 +32,70 @@ public final class Observable<T> {
...
             .store(in: &cancellables)
     }
+    
+    /// Binds the observable value to the values received from a Bindable's Publisher.
+    /// Changes to the observable value do not update the Bindable's value.
+    /// - Parameters:
+    ///   - bindable: A bindable object to receive values from.
+    ///   - dispatchQueue: The dispatchQueue on which to receive elements from the publisher.
+    ///   - animateUpdates: Animate value changes if the Bindable's control type supports it.
+    public func oneWayBind<B: Bindable>(with bindable: B,
```
Comment (nProdanov) [inline SFBaseKit/Utils/Observable.swift]: Do we actually need `animateUpdates`?

File: SFBaseKit/Utils/Observable.swift [hunk 4]
```diff
@@ -32,12 +32,70 @@ public final class Observable<T> {
     }
     
     /// Attaches a closure to execute when the binding value changes.
-    /// - Parameter receiveValue: The closure to execute with the new value.
+    /// The closure is executed on attachment.
+    /// - Parameters:
+    ///   - dispatchQueue: dispatchQueue on which to receive elements from the publisher.
```
Comment (ASPetrov) [inline SFBaseKit/Utils/Observable.swift:37]: [optional]

Maybe add "Default is `main`." at the end of the line.

P.S. Similar for any parameter with a default value,

**Review by ASPetrov — APPROVED** [review]

**Review by nProdanov — APPROVED** [review]
Great job!

File: SFBaseKit/Utils/Observable.swift (hunk 4 above)
Comment (NNikolov1415) [reply SFBaseKit/Utils/Observable.swift:37]: Added where applicable.

File: SFBaseKit/Utils/Observable.swift (hunk 1 above)
Comment (NNikolov1415) [reply SFBaseKit/Utils/Observable.swift]: Refactored into `oneWayBind` and `twoWayBind`.

File: SFBaseKit/Utils/Observable.swift (hunk 3 above)
Comment (NNikolov1415) [reply SFBaseKit/Utils/Observable.swift]: Removed.

**Pull request merged**